src/lib.rs

```diff
@@ -517,6 +600,39 @@ mod tests {
         assert_eq!(x.find_gte(257), None);
     }
 
+    #[test]
+    fn check_into_iter() {
+        let expected = vec![1, 2, 4, 8, 16, 32, 64, 128, 256];
+        let mut values = OrderedCollection::from_sorted_iter(expected.clone())
+            .into_iter()
+            .collect::<Vec<_>>();
+        values.sort();
+        assert_eq!(values, expected);
+    }
+
+    #[test]
+    fn check_into_iter_empty() {
+        let values = OrderedCollection::<u32>::from(vec![]);
+        assert!(Vec::from(values).is_empty());
+    }
+
+    #[test]
+    fn check_iter() {
+        let expected = vec![1, 2, 4, 8, 16, 32, 64, 128, 256];
+        let mut values = OrderedCollection::from_sorted_iter(expected.clone())
+            .iter()
+            .copied()
+            .collect::<Vec<_>>();
+        values.sort();
+        assert_eq!(values, expected);
+    }
+
+    #[test]
+    fn check_iter_empty() {
+        let values = OrderedCollection::<u32>::from(vec![]);
+        assert!(values.iter().next().is_none());
```

Comment:
```suggestion
        assert_eq!(values.iter().next(), None);
```
Gives slightly more helpful errors. Ditto in the other test.

Reply:
done